server/src/main/java/org/elasticsearch/search/aggregations/DelayedBucket.java

```diff
@@ -36,21 +34,15 @@ public final class DelayedBucket<B extends InternalMultiBucketAggregation.Intern
     /**
      * Build a delayed bucket.
      */
-    public DelayedBucket(
-        BiFunction<List<B>, AggregationReduceContext, B> reduce,
-        AggregationReduceContext reduceContext,
-        List<B> toReduce
-    ) {
-        this.reduce = reduce;
-        this.reduceContext = reduceContext;
+    public DelayedBucket(List<B> toReduce) {
         this.toReduce = toReduce;
     }
 
     /**
      * The reduced bucket. If the bucket hasn't been reduced already this
      * will reduce the sub-aggs and throw out the list to reduce.
      */
-    public B reduced() {
+    public B reduced(BiFunction<List<B>, AggregationReduceContext, B> reduce, AggregationReduceContext reduceContext) {
```

Comment:
Probably worth the same javadoc.

Reply:
Added and add an assert on `reduceContext.isFinalReduce()`